stdlib/public/SwiftOnoneSupport/SwiftOnoneSupport.swift

```diff
@@ -67,6 +132,14 @@ internal enum _Prespecialize {
       // force specialization of append.
       a.append(a[0])
 
+      #if _runtime(_ObjC)
+      // Explicitly specialize some private functions.
```

Comment:
Which ones are actually private? We should definitely not be emitting public pre-specializations of any; I would suggest removing any private functions from the list and only adding them back if we know we broke something.

However looking at a few, eg the `_ArrayBuffer` members, I don't think they're actually private at the SIL level. `_ArrayBuffer` is `@usableFromInline internal` and it has a bunch of `@inlinable internal` members. This means that those members are not visible in the source language but can be linked against publicly from SIL that has been deserialized into the client binary. So they're emitted with public linkage in the stdlib.

Do we know why we need special code to ensure that they're pre-specialized here?

Reply:
You are right, theses are not real "private" symbols, but @inlinable internal. Otherwise we wouldn't be able to specialize them.

We eagerly specialize a whole call tree, starting from a public function (referenced from the SwiftOnoneSupport.swift code). But because we compile the stdlib with generic inlining, it's not guaranteed that we see the complete (not-inlined) original call tree when specializing.

Reply:
It's still not entirely clear to me why we need two lists of symbols in two different places though. Perhaps we can just resolve this if you update the comment with your explanation, making it clear that we're not actually specializing private symbols (which would impact future library evolution).

Reply:
ok, I'll update the comment.